### PR TITLE
Create Paths and Models

### LINKED ISSUE #2: Database Schema
### User
- Email (string)
- Password hash (string)
- name (string)
- member_id (int)
- usertype (user or admin)
- timestamp (timestamp)

### Profile
- last_modified (timestamp)
- last_created (time_stamp)
- name (string)
- category (json list of category ids)
- offers_private_lessons (boolean)
- address (string)
- bio (string)
- website (string)
- phone (string)
- profile_img_url (string)
- social (json)
```
[{
  type: "twitter",
  url: "http://www.twitter.com/username"
}, ... ]
```
### Category
- id (int)
- name (string)

### PR DESCRIPTION
**Pull request by drewatk**
The initial models. With some caveats:

- I'm not sure how to store passwords
- Along with that, I'm not sure how to integrate our User into, Django's user management, nor weather we want to
- How can we do validation for the JSON fields

This resolves #2

### CODE REVIEW

**Comment by jonahheeren** [comment]
Can we just break the JSON fields into models with all optional fields? It doesn’t have to be represented in the DB as a separate table, but it still provides validation before committing it into the DB.

File: directory/urls.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.urls import path
+
+from . import views
+urlpatterns = [
+    path('', views.index, name='index'),
+]
```
Comment (jonahheeren) [inline directory/urls.py]: NIT: Newline

**Comment by drewatk** [comment]
Django does create every model as a table and generates all the SQL for that, so it wouldn't be the end of the world. Wouldn't it just be one extra table with a foreign key relationship to Profile?

**Comment by jonahheeren** [comment]
Yep, allowing nulls for every other field but the FK. The only reason it would really matter is if we don’t display the raw link. In which case I don’t relish the idea of a user clicking a link that says Facebook for an org and it leads anywhere.

**Comment by jonahheeren** [comment]
Not only that, but I could defs do an xss attack in a scenario where we allow: 

`<button id=“Facebookbutton”><a href=INPUT /> </button>`

Not that I think this scenario is likely, but it’s still good practice to guard against.

Forgive my shit html formatting, I’m doing this from my phone.

**Comment by drewatk** [comment]
It could be a one to many like this to avoid any NULLs?
screen shot 2018-02-15 at 3 24 16 pm

**Comment by jonahheeren** [comment]
If Postgres supports that without a reference table, I’m in.

**Comment by drewatk** [comment]
After looking in to it, we don't need our own user class, as Django's has all that we need. 

To manage permissions, in the shell or in a [fixture](https://docs.djangoproject.com/en/2.0/howto/initial-data/) we'll have to  [add users to permission groups](https://docs.djangoproject.com/en/2.0/ref/contrib/auth/#django.contrib.auth.models.Group).

**Review by jonahheeren — APPROVED** [review]

**Review by thesephist — APPROVED** [review]
LGTM.

One thing with migrations. Django's manage.py command generates migrations with generic `auto_*` names, but as long as the 4 digit number is kept the same you're actually free to modify the name of each migration file (before running it). Naming migration scripts makes it easier to tell in the future what it was for. i.e. `0002_update_profile_fields.py` is more descriptive than `0002_auto_[whatever].py`. We're probably fine for this batch of changes but just an FYI.

**Pull request merged**